confgenerator/logging_processors.go

```diff
@@ -364,3 +368,66 @@ end`, lua))...)
 func init() {
 	LoggingProcessorTypes.RegisterType(func() LoggingProcessor { return &LoggingProcessorExcludeLogs{} })
 }
+
+// A LoggingProcessorGceMetadataAttributes annotates logs with labels
+// corresponding to instance attributes from the GCE metadata server.
+type LoggingProcessorGceMetadataAttributes struct {
+	ConfigComponent `yaml:",inline"`
+	Include         []string `yaml:"include" validate:"dive,gt=0,glob"`
+	Exclude         []string `yaml:"exclude" validate:"dive,gt=0,glob"`
+	Static          *bool `yaml:"static" validate:"required,eq=true"`
+}
+
+func (p LoggingProcessorGceMetadataAttributes) Type() string {
```

Comment:
This doesn't need a `Type()` since it isn't being used in pipelines (yet?)

Reply:
I'd like to ensure that all `LoggingProcessor*` structs conform to the `LoggingProcessor` interface, which does require a `Type()`.

Reply:
I specifically want them to *not* conform to the interface to prevent them from accidentally leaking into external configs.

Reply:
That's what the "internal:" prefix check is for… Ideally, we would be able to represent most of the configs as Ops Agent pipelines (whether or not visible to the user), so making them conform to the interface makes sense to me.

Reply:
I don't think adding that prefix really changes anything. If you really care about having a valid processor, you already have one in the form of `LoggingProcessorModifyFields`, and you can just make this a factory function that returns a `LoggingProcessorModifyFields`.

```go
type addGceMetadataAttributes struct {
  Include, Exclude []string
}
 func (p addGceMetadataAttributes) Processor(ctx context.Context) LoggingProcessor {
  // body identical
  return LoggingProcessorModifyFields{
		Fields: modifications,
  }
}
```
Also, put it in a separate file (maybe `internal_processors.go`) so it's not commingled with user-facing processors.

Reply:
Removed the processor code.